### PR TITLE
feature/PDPDEVTOOL-3369: add customVmOptions to SdkExecutor

### PR DESCRIPTION
**Pull request by davidecorreu**
add customVmOptions to java-cli call
rename settings file to suitecloud-sdk-settings.json
add validation for custom vm options
add formating to .json generated files
improve messages related with settings file problems
fix error message for config:proxy command

### CODE REVIEW

File: packages/node-cli/src/SdkExecutor.js [hunk 1]
```diff
@@ -151,6 +152,17 @@ module.exports = class SdkExecutor {
 		return cliParamsAsString;
 	}
 
+	_getCustomVmOptionsString() {
+		const customVmOptions = this._CLISettingsService.getCustomVmOptions();
+		if (!customVmOptions) {
+			return '';
+		}
+		// customVmOptions are already validated at CLISettingsService 
+		return Object.keys(customVmOptions).reduce((acc, currentKey) =>
```
Comment (antoniofruci) [inline packages/node-cli/src/SdkExecutor.js:161]: I find the anonymous function hard to read. Extracting it and giving a name could improve code readability

File: packages/node-cli/src/services/settings/CLISettings.js [hunk 2]
```diff
@@ -4,11 +4,26 @@
...
+const DEFAULT_CLI_SETTINGS = {
+	useProxy: false,
+	proxyUrl: '',
+	isJavaVersionValid: false,
+}
+
 module.exports = class CLISettings {
+	// should not be directly used, use static fromJson(json) method
 	constructor(options) {
-		this._useProxy = options.useProxy;
```
Comment (lfpvillegas) [inline packages/node-cli/src/services/settings/CLISettings.js:9]: I think you should enforce that no one use this constructor then...
I'm looking at solutions that doesn't convince myself. We could look at it tomorrow.

https://stackoverflow.com/questions/21667149/how-to-define-private-constructors-in-javascript

**Review by lfpvillegas — CHANGES_REQUESTED** [review]

File: packages/node-cli/messages.json [hunk 3]
```diff
@@ -261,6 +260,9 @@
 	"ERRORS_SDKEXECUTOR_NO_TBA_FOR_ACCOUNT_AND_ROLE": "The email, NetSuite domain, account ID, and role combination in your account.json file has not been set up for token-based authentication (TBA). Run \"suitecloud account:setup\" to set up TBA.",
 	"ERRORS_SDKEXECUTOR_RUNNING_COMMAND": "There was an internal error while running the command. Details: {0}",
 	"ERRORS_SDKEXECUTOR_SDK_ERROR": "ERROR: SDK exited with code {0}\n{1}",
+	"ERRORS_SDK_SETTINGS_FILE_WRONG_JSON_SYNTAX": "Your {0} file is not valid. Ensure the content of the file is correct, or delete the file.\n{1}",
+	"ERRORS_SDK_SETTINGS_FILE_INVALID_CUSTOM_VM_OPTION": "There is an error with the \"vmOptions\" you set up in {0}.\n{1}",
```
[the remark this answers is not in the document]
Comment (davidecorreu) [reply packages/node-cli/messages.json:264]: This is the first message about vmOptions, there wasn't any before.
I'm using the " (double quotes), to be consistent with the other messages, and to refer to a property key in the settings file. (actually the message is from Alba, sniffff)

[the remark this answers is not in the document]
Comment (lfpvillegas) [reply packages/node-cli/messages.json:264]: Yeah I was referring to the message from the JIRA story. 
I thnk it's ok as you explained.

File: packages/node-cli/src/SdkExecutor.js [hunk 4]
```diff
@@ -91,7 +91,7 @@ module.exports = class SdkExecutor {
 	}
 
 	_launchJvmCommand(executionContext) {
-		if (!this._CLISettingsService.isJavaVersionValid()) {
+		if (this._CLISettingsService.isJavaVersionValid() !== true) {
```
[the remark this answers is not in the document]
Comment (davidecorreu) [reply packages/node-cli/src/SdkExecutor.js:94]: I like the idea, to move it to the getter. I'll apply it in the next commit.
I have done the change actually to enforce that the "isJavaVersionValid" value in the settings file to be a strictly a boolean. Before introducing this change a `123` value will be considered also a truly value.

[the remark this answers is not in the document]
Comment (davidecorreu) [reply packages/node-cli/src/SdkExecutor.js:94]: response in the comment above. (`123` will be considered as a truly value)

File: packages/node-cli/src/SdkExecutor.js (hunk 1 above)
Comment (davidecorreu) [reply packages/node-cli/src/SdkExecutor.js:161]: Agree. I'll give it a name for better readability .

File: packages/node-cli/src/SdkExecutor.js [hunk 5]
```diff
@@ -151,6 +152,17 @@ module.exports = class SdkExecutor {
...
 	}
 
+	_getCustomVmOptionsString() {
+		const customVmOptions = this._CLISettingsService.getCustomVmOptions();
+		if (!customVmOptions) {
+			return '';
+		}
+		// customVmOptions are already validated at CLISettingsService 
+		return Object.keys(customVmOptions).reduce((acc, currentKey) =>
+			acc+= customVmOptions[currentKey] === '' ? ` ${currentKey}` : ` ${currentKey}="${customVmOptions[currentKey].trim()}"`
```
[the remark this answers is not in the document]
Comment (davidecorreu) [reply packages/node-cli/src/SdkExecutor.js:162]: I'll apply the formatter.

**Review by lfpvillegas — APPROVED** [review]

**Pull request merged**